src/utils/MarkedExecution.ts

```diff
@@ -22,34 +22,37 @@ limitations under the License.
  * The function starts unmarked.
  */
 export class MarkedExecution {
-    private marked = false;
+    private _marked = false;
```

Comment:
final nitpick: now that we're not using the property, it should be named without the underscore. 

```suggestion
    private marked = false;
```